### PR TITLE
Wsba participant completion simple

### PR DESCRIPTION
**Pull request by npfr**
This quickstart is one that I've been planning to do for XTS. It's not in the list of AS71 quickstarts, but I'm happy for you to take it if you want it.

It uses technology that will ship and be supported in EAP 6.0.

### CODE REVIEW

File: wsba-participant-completion-simple/pom.xml [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+            <artifactId>jbossjta</artifactId>
+            <version>4.16.0.Beta1</version>
+            <scope>provided</scope>
+            <exclusions>
+                <exclusion>
+                    <groupId>org.jboss.logging</groupId>
+                    <artifactId>jboss-logging-spi</artifactId>
+                </exclusion>
+            </exclusions>
+        </dependency>
```
Comment (pmuir) [inline wsba-participant-completion-simple/pom.xml:98]: Don't we need the JBoss Repository defined to get these artifacts?

File: wsba-participant-completion-simple/pom.xml [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+            <version>1.0.0.CR4</version>
+            <scope>test</scope>
+        </dependency>
+
+    </dependencies>
+
+    <build>
+        <!-- Set the name of the war, used as the context root when the app
+      is deployed -->
+        <finalName>jboss-as-ejbinwar</finalName>
```
Comment (pmuir) [inline wsba-participant-completion-simple/pom.xml]: Make this ${project.artifactId} then it won't be wrong ;-)

File: wsba-participant-completion-simple/pom.xml [hunk 3]
```diff
@@ -0,0 +1,255 @@
...
+        <plugins>
+            <plugin>
+                <artifactId>maven-war-plugin</artifactId>
+                <version>2.1.1</version>
+                <configuration>
+                    <!-- Java EE 6 doesn't require web.xml, Maven needs to catch
+                   up! -->
+                    <failOnMissingWebXml>false</failOnMissingWebXml>
+                </configuration>
```
Comment (pmuir) [inline wsba-participant-completion-simple/pom.xml]: We can remove this configuration now as you do need a web.xml

**Comment by pmuir** [comment]
I'm happy to include this, pending Sande's usability review of course. If you could collapse it down into one commit before we merge that would be good!

**Comment by npfr** [comment]
FYI,

The commit list is currently looking a bit strange. I managed to break it whilst attempting a squash. I'm currently trying to fix it. 

This shouldn't prevent Sande from doing a usability review.

Paul.

**Comment by npfr** [comment]
OK, I think I've successfully squashed to one commit and also addressed your other comments.

**Comment by sgilda** [comment]
Hi Paul,
I'm a stupid user. I'm following the instructions in the README (which is excellent, btw), but I must be missing something. My understanding is that all I need to do is start the server and then run this: 
     mvn clean test -Parq-jbossas-remote.
When I do that, I get a BUILD FAILURE:
    Tests in error: 
       org.jboss.as.quickstarts.wsba.participantcompletion.simple.ClientTest: Could not deploy to container

I am running jboss-as-7.1.0.Final-SNAPSHOT.
Did I miss a step or misunderstand what I need to do?

Thanks,
Sande

**Comment by npfr** [comment]
Hi Sande,

The problem was that it was still configured to use JBossAS-7.1.0.Beta1. I've upgraded to CR1b and also made updates to the Arquillian version too. This should fix it. I think there are other quickstarts that have the same problem. I'll take a look and raise a pull req if appropriate.

I'll squash into a single commit, once it's been approved.

Paul.

**Comment by npfr** [comment]
Looks like the problem occurs prior to 7.1.0.Beta1b and is probably caused by the remoting incompatibility that necessitated the 7.1.0.Beta1b release.

I've checked master and all Arquillian versions are using AS 7.1.0.Beta1b or above:

```
[paul@wiggum quickstart]$ grep -r -A 1 "<artifactId>jboss-as-arquillian-container-remote</artifactId>" .
./html5-mobile/pom.xml:               <artifactId>jboss-as-arquillian-container-remote</artifactId>
./html5-mobile/pom.xml-               <version>7.1.0.Beta1b</version>
--
./kitchensink/pom.xml:               <artifactId>jboss-as-arquillian-container-remote</artifactId>
./kitchensink/pom.xml-               <version>7.1.0.Beta1b</version>
--
./kitchensink-ear/jboss-as-kitchensink-ear-ejb/pom.xml:               <artifactId>jboss-as-arquillian-container-remote</artifactId>
./kitchensink-ear/jboss-as-kitchensink-ear-ejb/pom.xml-               <version>7.1.0.Beta1b</version>
--
./kitchensink-jsp/pom.xml:               <artifactId>jboss-as-arquillian-container-remote</artifactId>
./kitchensink-jsp/pom.xml-               <version>7.1.0.Beta1b</version>
```

To be sure I ran the Arquillian tests on kitchensink against 7.1.0.CR1b and they pass.

I assume you have plans to control these version numbers centrally, so I won't bother issuing a pull request to update them.

**Comment by sgilda** [comment]
Thanks Paul.

Pete, do we control them centrally? Or do we expect them to match up for 
the scripts to find them?

**Comment by npfr** [comment]
I've applied the AS7 code style.

**Comment by npfr** [comment]
I've also moved the index entry for this quickstart into README.md

**Comment by npfr** [comment]
Squashed

**Comment by pmuir** [comment]
Sande, are we done on this one? Can we merge it?

**Comment by sgilda** [comment]
I now see BUILD SUCCESS with this result: Tests run: 3, Failures: 0, Errors: 0, Skipped: 0
However, I'm not clear what I'm supposed to see for the result.

in the server console I see that it starts the deployment, adds a service, registers it, then there are a lot of "Creating Service" messages. 
Then I see this exception in the log:
  Method invocation failed with exception: null: java.lang.reflect.InvocationTargetException
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) [:1.6.0_25]
    {removed stacktrace}
  Caused by: org.jboss.as.quickstarts.wsba.participantcompletion.simple.AlreadyInSetException: 
        item '1' is already in the set.
Then I see more "Creating Service" messages, followed by "remove: jboss.ws:context=wsba,endpoint=SetServiceBA", then "Stopped deployment wsba.war" , then  Undeployed "wsba.war"

The README says it assumes you have an understanding of WS-BusinessActivity. I'm a stupid user, so I don't. My question is, what am I looking for when I run this example to know that this is running correctly? Should I see anything in shell prompt where I run "mvn clean test -Parq-jbossas-remote"? Do I look in the server log? If so, what should I expect to see? 

There is a minor typo 'tht' in the README: 'It is also assumed tht you'.

**Comment by npfr** [comment]
I added logging to the quickstart, but it is difficult to differentiate from all the other logging produced by JBossAS, XTS, JBossWS, etc. For that reason, I didn't mention it in the readme.

However, I agree it would be better to provide some indication of what is going on. I'll have a think about the best way to do this. I'll also take a look at some of the other quickstarts as I'm sure they will have hit the same issue.

**Comment by sgilda** [comment]
Thanks. I appreciate it!

**Comment by npfr** [comment]
Added the expected output and improved the server startup instructions. @sgilda, can you re-review please?

**Comment by npfr** [comment]
@sgilda I've fixed the inconsistencies.

**Comment by sgilda** [comment]
Thanks. I'll take a look at it now.

**Comment by sgilda** [comment]
This looks great! I will merge it in soon.

**Comment by sgilda** [comment]
Hi Paul,
While copying a description for the quickstart page, I noticed a small typo "Bushiness" in the README file:
This example demonstrates the deployment of a WS-BA (WS Bushiness Activity) enabled JAX-WS Web service bundled in a war archive for deployment to JBoss AS 7.
Thought you might want to change it on your end. :-)
Sande

**Comment by npfr** [comment]
Cheers, I'll get that sorted tomorrow.

**Comment by npfr** [comment]
Fixed the 'bushiness' typo.

**Comment by npfr** [comment]
I think I have the version numbers aligned with HEAD.
I changed 'arquillian-junit-container' and 'arquillian-protocol-servlet'.

**Comment by npfr** [comment]
Squashed

**Comment by sgilda** [comment]
Looking at this next...

**Comment by sgilda** [comment]
Hi Paul,
Sorry for the delay in getting to this. 
I just looked at your POM and noticed some of the versions are off a bit. For example:
jboss-as-arquillian-\* artifactId patterns should be 7.1.0.CR1b
org.jboss.arquillian.\* ﻿groupIds pattern should be 1.0.0.CR7.
jboss-as-maven-plugin artifactId should be 7.1.0.CR1 
This information was probably changed after you started this quickstart. We need them all to be consistent so we can use scripts to update them when they are moved to EAP and also to avoid runtime errors.
I created this page to use for a reference: https://docs.jboss.org/author/display/AS71/Maven+POM+Versions+Checklist 
Sorry to make you go through all this again, but do you mind updating the pom.xml file again? 
Thanks, Sande

**Comment by sgilda** [comment]
I made the changes and issued pull 81: https://github.com/jbossas/quickstart/pull/81.
Pete, is it OK to close this one?

**Comment by pmuir** [comment]
81 is closed. Closing this one too.

**Pull request closed without merging**